pkg/cmd/server/kubernetes/master.go

```diff
@@ -380,12 +315,14 @@ func (c *MasterConfig) RunGarbageCollectorController(client *osclient.Client, co
 
 	config = restclient.AddUserAgent(config, "generic-garbage-collector")
 	config.ContentConfig.NegotiatedSerializer = serializer.DirectCodecFactory{CodecFactory: metaonly.NewMetadataCodecFactory()}
+	// TODO: should use a dynamic RESTMapper built from the discovery results.
+	restMapper := registered.RESTMapper()
```

Comment:
does this increase our exposure to garbage collection getting confused by origin resources (which it won't be able to find)?